### PR TITLE
docs: document process experience

### LINKED ISSUE #12: `P8` Document process and experience
Description: The overall work carried out, and experience gained, are documented. In particular, mention your experience about the given documentation/examples of the project, its tool framework, and the interaction within your team (using the Essence framework) and with the community of the project.

Definition of done: This section is added and merged to the `report` branch

### PR DESCRIPTION
**Pull request by lskare**
Add answers to pass criteria P8

Includes answers on how our group:
- has progressed w.r.t Essence
- has worked during the assignment
- main take-aways
- interaction with the pdm community
- pdms built in tools and documentation

### CODE REVIEW

File: Fundsoft/report.md [hunk 1]
```diff
@@ -127,10 +127,44 @@ Optional (point 2): relation to design pattern(s).
...
+
+### Distribution of work 
+For this assignment we had two start-up meetings where we discussed the issue, how we roughly wanted to tackle it along with creating specific atomic issues that could be handled individually. After these were all created on GitHub some members of the group starting looking at solving the issue meanwhile others started working on tests for the issue. At the same time we discussed how we wanted to structure the report and create more atomic issues related to this. As we were already familiar with the pdm library we could effectively do all these things simultaneously and get to work quite quickly with little extra onboarding. It was also beneficial to do the tests and solving the issue at the same time in order to see how well they coincided with each other and allowed us to continuosly assess our patch.
```
Comment (Stenmarken) [inline Fundsoft/report.md]: *while others started working on tests

File: Fundsoft/report.md [hunk 2]
```diff
@@ -127,10 +127,44 @@ Optional (point 2): relation to design pattern(s).
...
+
+### Main take-aways
 What are your main take-aways from this project? What did you learn?
 
+* Importance of clear issue descriptions and conveying ones ideas when working on an open source project
+    * As anyone can attempt to contribute to an open source project it is very important to be extra clear with the desired functionalities of ones issues as people will not be fully emersed in the details of the project
+    * We were very thankful for the clear description of the issue along with fast response times from the open source manager. This helped us throughly understand the problem and distribute the work.
+* Reinforce the importance of systematically organizing the group in order to tackle the problem efficiently
+    * As this was a very open problem we felt that we really benefited from having start-up meetings aimed to clearly deconstruct the problem into atomic issues and spread the workload.
+* We've learnt how to create a final patch that can be easily merged into an open source project and that is clean and almost surgecal in the sense that it does not negatively affect the exiting project.
```
Comment (Stenmarken) [inline Fundsoft/report.md]: *surgical

File: Fundsoft/report.md [hunk 3]
```diff
@@ -127,10 +127,44 @@ Optional (point 2): relation to design pattern(s).
...
+* It is emmensely important to create and adhere to guidelines when contributing to an open source project. Otherwise you risk merging many different styles from potentially hundreds of small contributors which can result in a chaotic mess.
```
Comment (Stenmarken) [inline Fundsoft/report.md]: * immensely

File: Fundsoft/report.md [hunk 4]
```diff
@@ -127,10 +127,44 @@ Optional (point 2): relation to design pattern(s).
...
 ## Overall experience
 
+Overall work
+
+Our contribution to the open source library pdm can be summarized as the following:
+
+### Chosen issue and open source interaction
+Chosen issue from pdm which was assigned to Rasmus (rasmus-d): https://github.com/pdm-project/pdm/issues/2628 
+
+This issue intends to tackle a problem with pdm where a user cannot update a single sub-dependency. Before our patch is was only possible for a user to update a single dependency, all dependencies or all dependencies + all subdependencies but not a single sub-dependecy. Thus the issue relates to implementing a feature where a user can update a specific sub-dependency without changing/updating any other dependency or subdependency.
```
Comment (Stenmarken) [inline Fundsoft/report.md]: Isn't it possible also to update a dependency + its sub-dependencies?

**Review by Stenmarken — COMMENTED** [review]
Really good work Ludwig! Your writing is easy to read yet and is just the right length.

File: Fundsoft/report.md (hunk 4 above)
Comment (lskare) [reply Fundsoft/report.md]: Yes it is! Good catch

**Review by Stenmarken — APPROVED** [review]
Really good work!

**Pull request merged**